### DISCUSSION TITLE
Convert JP2 to HEIC

### DISCUSSION THREAD

**Discussion by patoperro** [ImageMagick/ImageMagick#2594] [Help]
Hi all, 

I'm new to imagemagick, which I just installed via MacPorts. (In the protocol I found the line "x ./opt/local/lib/ImageMagick-6.9.11/modules-Q16/coders/heic.so" so can I assume all the right plugins are installed to work with HEIC images?)

**The aim**
I want to convert JP2 files to HEIC.

**The why**
I have a ton of JPEG2000 files which I decided to used back when scanning 35mm and Medium format negatives to save disk space. Lightroom doesn't work with JP2 and I am having difficulties getting the files to load in digikam (which says they cannot be loaded and simply displays a generic thumbnail). What I would like to do is batch convert them to HEIC, as this would allow me to at least browse through my image catalogue.

**The how**
Can anyone tell me if this is going to work using imagemagick and if so, then point me in the right direction?

Thanks,
Pato

**Comment by fmw42** [comment]
I believe that you need to install libheif. See HEIC in the list of formats supported at https://imagemagick.org/script/formats.php. But MacPorts may do that for you. You can see if it is installed if you run (assuming IM 6)

`convert -version `

and you see heic in the list of Delegates.

**Reply by patoperro** [reply]
Thanks for that tip. HEIC is listed in the delegates.

Then is it only a question of batch convert syntax, I suppose. 
I read that there are certain difficulties with color profiles, but didn't grasp what should be done to avoid such issues.

**Reply by fmw42** [reply]
Sorry, I am not sure about the profiles issue. I recall some discussions. Try one convert of a file to heic and see if it works as needed. Also search the Discussions and Bug Report forums and see what was said there.

**Comment by snibgo** [comment]
> I read that there are certain difficulties with color profiles, but didn't grasp what should be done to avoid such issues.

I haven't heard of any difficulties. All HEIC files I have seen have an embedded ICC profile, Apple's "Display P3". If you want to convert to another colorspace, use IM's "-profile" operation.

**Reply by patoperro** [reply]
I used "eciRGB_v2_ICCv4.icc" in all of my scans (.jp2 files) and want to keep it that way in the HEIC files. I'll have to look more deeply at the documentation.
